modules/videoio/test/test_camera.cpp

```diff
@@ -72,6 +72,27 @@ TEST(DISABLED_videoio_camera, basic)
     capture.release();
 }
 
+// Test that CAP_PROP_CONVERT_RGB remain to false (default is true) after other supported property are set.
+// The test use odd value to be almost sure to trigger code responsible for recreating the device.
+TEST(DISABLED_videoio_camera, dshow_convert_rgb_persistency)
+{
+    VideoCapture capture(CAP_DSHOW);
+    ASSERT_TRUE(capture.isOpened());
+    ASSERT_TRUE(capture.set(CAP_PROP_CONVERT_RGB, 0));
+    ASSERT_DOUBLE_EQ(capture.get(CAP_PROP_CONVERT_RGB), 0);
+    capture.set(CAP_PROP_FRAME_WIDTH, 641);
+    capture.set(CAP_PROP_FRAME_HEIGHT, 481);
+    capture.set(CAP_PROP_FPS, 31);
+    capture.set(CAP_PROP_CHANNEL, 1); 
+    capture.set(cv::CAP_PROP_FOURCC, cv::VideoWriter::fourcc('Y', '1', '6', ' '));
+    std::cout << "Camera 0 via " << capture.getBackendName() << " backend" << std::endl;
+    std::cout << "Frame width: " << capture.get(CAP_PROP_FRAME_WIDTH) << std::endl;
+    std::cout << "     height: " << capture.get(CAP_PROP_FRAME_HEIGHT) << std::endl;
+    std::cout << "Capturing FPS: " << capture.get(CAP_PROP_FPS) << std::endl;
+    ASSERT_DOUBLE_EQ(capture.get(CAP_PROP_CONVERT_RGB), 0);
+    capture.release();
```

Comment:
Please add `lastFrame` check (as proposed above).
Look like `CAP_PROP_CONVERT_RGB=0` + `CAP_PROP_FOURCC=Y16` still doesn't work: .type() is 8UC3, frame is in color.

---

You can add this for local experiments:
```
imshow("test", lastFrame);
waitKey();
```

Reply:
This is not the bug this PR is correcting. 
This PR is correcting the fact that CONVERT_RGB once set to `false` should stay false no matters if width/height/fourcc/fps is set after.
This is what my test is doing. This is what my PR is merging. This is not related to the fact i'm opening my camera in Y16.

*If you look implementation, it seems like `convertRgb` is used only with `MEDIASUBTYPE_Y16`, so yes it is linked a little bit with Y16.*

The fact that `CAP_PROP_CONVERT_RGB=0` + `CAP_PROP_FOURCC=Y16` doesn't give 16U mat is another bug. If the camera support being opened in Y16 mode, then `CV_16UC1 == lastFrame.type()`

I tested with FLIR Boson320 i can confirm, the read frame is CV_16UC1.
When opening my webcam with the same code (just changing the id), the result mat is `CV_8UC3`.

This might be another bug, but this is unrelated to what my PR fix.

The code that doesn't open Y16 if camera doesn't support is there i believe:
https://github.com/opencv/opencv/blob/900051cd04b14806c84b0a45f0dd66bb0f53ad36/modules/videoio/src/cap_dshow.cpp#L2616-L2623

From reading source code it also seems that `CAP_PROP_CONVERT_RGB=1` + `CAP_PROP_FOURCC=Y8` will give you `CV_8UC1` as output 

https://github.com/opencv/opencv/blob/900051cd04b14806c84b0a45f0dd66bb0f53ad36/modules/videoio/src/cap_dshow.cpp#L3562-L3569

Reply:
> This might be another bug, but this is unrelated to what my PR fix.

OK.

---

Please fix remaining coding style issue (check "Docs" builder logs) before merge.

Reply:
I'm not sure to understand, where in Docs builder? I'm not familiar with Jenkins.

Reply:
"default" entry: http://pullrequest.opencv.org/#/summary/opencv